torch/csrc/jit/codegen/cuda/lower_trivial_broadcast.cpp

```diff
@@ -96,27 +96,27 @@ void ConcretizedBroadcastDomains::handle(Expr* expr) {
 }
 
 void ConcretizedBroadcastDomains::markAsConcretized(IterDomain* root_domain) {
-  concretized_domains_.emplace(root_domain);
   std::deque<IterDomain*> child_domains({root_domain});
+  std::unordered_set<IterDomain*> visited;
   while (!child_domains.empty()) {
     auto child = child_domains.front();
     child_domains.pop_front();
+    if (!concretized_domains_.emplace(child).second) {
+      continue;
+    }
     const auto& child_uses = child->uses();
-    if (child_uses.size() == 1 && !child_uses[0]->outputs().empty() &&
-        std::all_of(
-            child_uses[0]->outputs().begin(),
-            child_uses[0]->outputs().end(),
-            [](const Val* out) { return out->isA<IterDomain>(); })) {
-      for (auto out : child_uses[0]->outputs()) {
-        child_domains.push_back(out->as<IterDomain>());
+    for (auto child_use : child_uses) {
+      for (auto out_id :
+           ir_utils::filterByType<IterDomain>(child_use->outputs())) {
+        if (visited.find(out_id) == visited.end()) {
```

Comment:
It seems `visited` never gets inserted. This seems to continue traversing as long as `IterDomain` is found in the outputs of `child_use`.